### PR TITLE
End to End Test cases for - installing package from a local source that doesn't exist throws a bogus message

### PR DESCRIPTION
**Pull request by mishra14**
Adding test cases for -

bug : https://github.com/NuGet/Home/issues/1674

original code PR's : https://github.com/NuGet/NuGet.Client/pull/742 https://github.com/NuGet/NuGet.Client/pull/758

Added End to End test cases to check the correctness of the error messages for the source specified using source switch.

cc : @alpaix  @rohit21agrawal @jainaashish @rrelyea @emgarten @joelverhagen

### CODE REVIEW

**Comment by alpaix** [comment]
looks good. :shipit:

File: test/EndToEnd/tests/install.ps1 [hunk 1]
```diff
@@ -1,5 +1,116 @@
...
+	$package = "Rules"
+	$project = New-ConsoleApplication
+	$source = "..\"
+	$message = "Unable to find package '$package'"
+
+	# Act & Assert
+	Assert-Throws { Install-Package $package -ProjectName $project.Name -source $source } $message
+}
+
+function Test-InstallPackageWithInvalidHttpSource {
```
Comment (alpaix) [inline test/EndToEnd/tests/install.ps1:48]: Please add one more with `-Verbose`

Comment (mishra14) [inline test/EndToEnd/tests/install.ps1:48]: Added. Please take a look.

File: test/EndToEnd/tests/install.ps1 [hunk 2]
```diff
@@ -56,6 +56,23 @@ function Test-InstallPackageWithInvalidHttpSource {
...
 
+function Test-InstallPackageWithInvalidHttpSourceVerbose {
+	# Arrange
+	$package = "Rules"
+	$project = New-ConsoleApplication
+	$source = "http://example.com"
+	$url = $source+"/FindPackagesById()?id='$package'"
+	$errorcode = "404 Not Found"
+	$message = "  GET $url   NotFound $url ? An error occurred while retrieving package metadata for '$package' from source '$source'."+
+               "  The V2 feed at '$url' returned an unexpected status code '$errorcode'. Unable to find package 'Rules' at source '$source'."
```
Comment (alpaix) [inline test/EndToEnd/tests/install.ps1]: escape dots: . => \.

File: test/EndToEnd/tests/install.ps1 [hunk 3]
```diff
@@ -56,6 +56,23 @@ function Test-InstallPackageWithInvalidHttpSource {
...
+	
+	# Act
+	$result = Install-Package $package -ProjectName $project.Name -source $source -Verbose *>&1
+
+	# Assert
+	Assert-True { $message -match $result }
```
Comment (alpaix) [inline test/EndToEnd/tests/install.ps1]: `$result -match $message`

File: test/EndToEnd/tests/install.ps1 (hunk 2 above)
Comment (alpaix) [inline test/EndToEnd/tests/install.ps1]: Actually it'd be better to escape the whole thing altogether with `[regex]::Escape($message)`

**Comment by alpaix** [comment]
:shipit:

**Comment by mishra14** [comment]
Merged : https://github.com/NuGet/NuGet.Client/commit/018ebd6d57ff75572ee2055c428b51be30b8a176

**Pull request closed without merging**